### ISSUE TITLE
Make it easier to change the hyper-parameters in the optimizer 

### ISSUE THREAD

**Issue by user1823** [open-spaced-repetition/fsrs4anki#241]
It could be better to allow professional users to set the hyper-parameters easily.

_Originally posted by @L-M-Sherlock in https://github.com/open-spaced-repetition/fsrs4anki/issues/187#issuecomment-1501060463_

**Comment by user1823** [comment]
> It could be better to allow professional users to set the hyper-parameters easily.

With the latest update (the code being hidden away in a separate file), it becomes even more important to add an option to easily change the n_epoch and learning rate (and probably other factors also).

**Comment by L-M-Sherlock** [comment]
Could you provide a list of hyper-parameters that need to pass into the `train()` function?

**Comment by user1823** [comment]
Currently, I can only think of the above two (n_epoch and learning rate) because these are the only ones that I have tried changing till now.

**Comment by user1823** [comment]
@Expertium, can you think of some other factors that should be made easily adjustable?

**Comment by Expertium** [comment]
I can't think of anything else.
Optimizer types could be made easily adjustable (good old Stochastic Gradient Descent or its fancier versions like Adam, which is what Sherlock is currently using), but without a good understanding of how they work, users won't find this useful. Plus, I doubt that it will make much of a difference; as far as I know, the more sophisticated optimizers don't always outperform SGD.

**Comment by L-M-Sherlock** [comment]
In v3.18.1, you can call `optimizer.train()` with parameters `lr` and `n_epoch`.

**Comment by user1823** [comment]
But exactly how? I am not a developer. So, it is not that easy for me to figure out.

Wouldn't it make sense to allow the user to change it in the way they can change the other values like the `timezone`, `next_day_starts_at`, etc.?

**Comment by L-M-Sherlock** [comment]
Replace `optimizer.train()` with something like `optimizer.train(lr=0.005, n_epoch=3)`.
